### PR TITLE
feat: Added metrics for edit networks and accounts

### PR DESCRIPTION
**Pull request by NidhiKJha**
This PR is to add metrics for edit networks and accounts screen

## **Related issues**

Fixes: [https://github.com/MetaMask/MetaMask-planning/issues/3263](https://github.com/MetaMask/MetaMask-planning/issues/3263)

## **Manual testing steps**

1. Go to ui/contexts/metametrics.js
2. In trackEvent in this file, add a log to see the payload
3. Run extension with yarn start. Go to Permissions Page, click on edit button for accounts and networks and check the console to verify the payload. In edit modals, if we update accounts or networks. Check the payload as well

### **Before**

NA
### **After**

NA
## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/multichain/edit-accounts-modal/edit-accounts-modal.tsx [hunk 1]
```diff
@@ -181,7 +188,29 @@ export const EditAccountsModal: React.FC<EditAccountsModalProps> = ({
                     <ButtonPrimary
                       data-testid="connect-more-accounts-button"
                       onClick={() => {
+                        const addedAccounts = selectedAccountAddresses.filter(
```
Comment (andreahaku) [inline ui/components/multichain/edit-accounts-modal/edit-accounts-modal.tsx:191]: this should be a faster and more efficient approach:
```
// Convert the arrays to Sets for O(1) lookup
const defaultSet = new Set(defaultSelectedAccountAddresses);
const selectedSet = new Set(selectedAccountAddresses);

// Get accounts that are in `selectedAccountAddresses` but not in `defaultSelectedAccountAddresses`
const addedAccounts = selectedAccountAddresses.filter(
  (address) => !defaultSet.has(address),
);

// Get accounts that are in `defaultSelectedAccountAddresses` but not in `selectedAccountAddresses`
const removedAccounts = defaultSelectedAccountAddresses.filter(
  (address) => !selectedSet.has(address),
);
```

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.js [hunk 2]
```diff
@@ -180,6 +185,22 @@ export const EditNetworksModal = ({
                 data-testid="connect-more-chains-button"
                 onClick={() => {
                   onSubmit(selectedChainIds);
+                  const addedNetworks = selectedChainIds.filter(
```
Comment (andreahaku) [inline ui/components/multichain/edit-networks-modal/edit-networks-modal.js:188]: same approach here:
```
// Convert the arrays to Sets for O(1) lookup
const defaultChainIdsSet = new Set(defaultSelectedChainIds);
const selectedChainIdsSet = new Set(selectedChainIds);

// Get networks that are in `selectedChainIds` but not in `defaultSelectedChainIds`
const addedNetworks = selectedChainIds.filter(
  (chainId) => !defaultChainIdsSet.has(chainId),
);

// Get networks that are in `defaultSelectedChainIds` but not in `selectedChainIds`
const removedNetworks = defaultSelectedChainIds.filter(
  (chainId) => !selectedChainIdsSet.has(chainId),
);
```

**Review by andreahaku — COMMENTED** [review]
a couple of optimisation comments

**Review by andreahaku — APPROVED** [review]
LGTM

File: ui/components/multichain/edit-networks-modal/edit-networks-modal.js (hunk 2 above)
Comment (NidhiKJha) [reply ui/components/multichain/edit-networks-modal/edit-networks-modal.js:192]: Updated here https://github.com/MetaMask/metamask-extension/pull/27820/commits/69389a8f9e8d17afd7c48c5c3f3b10ca7847e070

File: ui/components/multichain/edit-accounts-modal/edit-accounts-modal.tsx (hunk 1 above)
Comment (NidhiKJha) [reply ui/components/multichain/edit-accounts-modal/edit-accounts-modal.tsx:195]: Updated here https://github.com/MetaMask/metamask-extension/pull/27820/commits/69389a8f9e8d17afd7c48c5c3f3b10ca7847e070

**Review by jonybur — APPROVED** [review]

**Pull request merged**